contracts/mothership/OverlayV1Mothership.sol

```diff
@@ -60,65 +60,59 @@ contract OverlayV1Mothership is AccessControlEnumerable {
 
         _setupRole(ADMIN, msg.sender);
         _setupRole(GOVERNOR, msg.sender);
-        _setupRole(GUARDIAN, msg.sender);
         _setRoleAdmin(GOVERNOR, ADMIN);
-        _setRoleAdmin(GUARDIAN, ADMIN);
-
-        // global params
-        fee = _fee;
-        feeBurnRate = _feeBurnRate;
-        feeTo = _feeTo;
-        marginBurnRate = _marginBurnRate;
-
-    }
-
-    function setOVL (address _ovl) external onlyGovernor {
 
+        // immutable params
         ovl = _ovl;
 
+        // global params
+        _setFeeTo(_feeTo);
+        _setFee(_fee);
+        _setFeeBurnRate(_feeBurnRate);
+        _setMarginBurnRate(_marginBurnRate);
     }
 
     function totalMarkets () external view returns (uint) {
         return allMarkets.length;
     }
 
+    function totalCollaterals () external view returns (uint) {
+        return allCollaterals.length;
+    }
 
     /**
       @notice Make this contract aware of a new market contract's existence
       @dev Should be called after contract deployment in specific market factory.createMarket
       @dev Only the Governor can initialize a market
       @dev Appends new market address to `allMarkets` array to track them
-      @param market Overlay market contract address
+      @param _market Overlay market contract address
       */
-    function initializeMarket(address market) external onlyGovernor {
-
-        require(!marketExists[market], "OVLV1:!!initialized");
-
-        marketExists[market] = true;
-        marketActive[market] = true;
+    function initializeMarket(address _market) external onlyGovernor {
```

Comment:
This isn't really initializing a market, that seems to say we are instantiating some state on the market. We are just adding it to the manifest here in the mothership.